### PR TITLE
simplify MapVote.Start logic

### PR DESCRIPTION
**Pull request by plally**

### CODE REVIEW

File: lua/mapvote/config.lua [hunk 1]
```diff
@@ -1,9 +1,52 @@
...
+            if decoded.maps[0] == "^" then
+                MapVote.Config.MapPrefixes = { string.sub(decoded.maps, 2) }
+            else
+                MapVote.Config.MapPrefixes = { decoded.maps }
+            end
+        end
+    else
+        ErrorNoHalt( "MapVote Prefix can not be loaded from gamemode" )
+    end
+end
```
Comment (wrefgtzweve) [inline lua/mapvote/config.lua]: no newline at the end of the file

File: lua/mapvote/sv_mapvote.lua [hunk 2]
```diff
@@ -87,127 +87,74 @@ local function mapVoteOver( autoGamemode, callback )
...
-                    RunConsoleCommand( "gamemode", gamemode )
-                end
-                RunConsoleCommand( "changelevel", map )
-            end
+        if hook.Run("MapVoteChange", map) == false then return end
+        if callback then
+            callback( map )
         end
+
+        RunConsoleCommand( "changelevel", map )
```
Comment (wrefgtzweve) [inline lua/mapvote/sv_mapvote.lua:96]: In the original code it runs either callback or the console command, in this pr it does both. Not sure if thats intentional

File: lua/mapvote/sv_mapvote.lua [hunk 3]
```diff
@@ -87,127 +87,74 @@ local function mapVoteOver( autoGamemode, callback )
...
-        is_expression = true
-    else
-        if prefix and type( prefix ) ~= "table" then prefix = { prefix } end
     end
+    return false
+end
 
-    local maps = file.Find( "maps/*.bsp", "GAME" )
+function MapVote.Start( length, callback )
+    length = length or MapVote.Config.TimeLimit or 28
```
Comment (wrefgtzweve) [inline lua/mapvote/sv_mapvote.lua:122]: not sure why we should keep this 28, might be nicer to have the fallback to a whole number like 30? turbo nitpick ofc

**Review by wrefgtzweve — COMMENTED** [review]
Some nitpicks

File: lua/mapvote/sv_mapvote.lua (hunk 2 above)
Comment (plally) [reply lua/mapvote/sv_mapvote.lua:96]: it is

Comment (plally) [reply lua/mapvote/sv_mapvote.lua:96]: this behavior makes more sense and we dont use the callback for anything anyway

File: lua/mapvote/sv_mapvote.lua (hunk 3 above)
Comment (plally) [reply lua/mapvote/sv_mapvote.lua:122]: that was the default before

File: lua/mapvote/sv_mapvote.lua [hunk 4]
```diff
@@ -87,127 +87,74 @@ local function mapVoteOver( autoGamemode, callback )
...
 end
 
-function MapVote.Start( length, current, limit, prefix, callback )
-    current = current or MapVote.Config.AllowCurrentMap or false
-    length = length or MapVote.Config.TimeLimit or 28
-    limit = limit or MapVote.Config.MapLimit or 24
-    prefix = prefix or MapVote.Config.MapPrefixes
-    local cooldown = MapVote.Config.EnableCooldown or MapVote.Config.EnableCooldown == nil and true
-    local autoGamemode = autoGamemode or MapVote.Config.AutoGamemode or MapVote.Config.AutoGamemode == nil and true
+local function isMapAllowed( m )
```
Comment (wrefgtzweve) [inline lua/mapvote/sv_mapvote.lua:100]: single letter variable name is quite vague, something like `map` might be better

**Pull request merged**